pkg/lib/cockpit-components-terminal-scribery.jsx

```diff
@@ -0,0 +1,201 @@
+     * stream channel.
+     *
+     * The size of the terminal can be set with the 'rows' and 'cols'
+     * properties. If those properties are not given, the terminal will fill
+     * its container.
+     *
+     * If the 'onTitleChanged' callback property is set, it will be called whenever
+     * the title of the terminal changes.
+     *
+     * Call focus() to set the input focus on the terminal.
+     */
+    var Terminal = React.createClass({
+        propTypes: {
+            cols: React.PropTypes.number,
+            rows: React.PropTypes.number,
+            channel: React.PropTypes.object.isRequired,
+            onTitleChanged: React.PropTypes.func
+        },
+
+        componentWillMount: function () {
+            var term = new Term({
+                cols: this.state.cols || 80,
+                rows: this.state.rows || 25,
+                screenKeys: true,
+                useStyle: true
+            });
+
+            term.on('data', function(data) {
+                if (this.props.channel.valid)
+                    this.props.channel.send(data);
+            }.bind(this));
+
+            if (this.props.onTitleChanged)
+                term.on('title', this.props.onTitleChanged);
+
+            this.setState({ terminal: term });
+
+            window.addEventListener("keydown", this.handleKeyDown, false);
+        },
+
+        componentDidMount: function () {
+            this.state.terminal.open(this.refs.terminal);
+            this.connectChannel();
+
+            if (!this.props.rows) {
+                window.addEventListener('resize', this.onWindowResize);
+                this.onWindowResize();
+            }
+        },
+
+        componentWillUpdate: function (nextProps, nextState) {
+            if (nextState.cols !== this.state.cols || nextState.rows !== this.state.rows) {
+                this.state.terminal.resize(nextState.cols, nextState.rows);
+                this.props.channel.control({
+                    window: {
+                        rows: nextState.rows,
+                        cols: nextState.cols
+                    }
+                });
+            }
+
+            if (nextProps.channel !== this.props.channel) {
+                this.state.terminal.reset();
+                this.disconnectChannel();
+            }
+        },
+
+        componentDidUpdate: function (prevProps) {
+            if (prevProps.channel !== this.props.channel)
+                this.connectChannel();
+        },
+
+        sendToTerm: function(value) {
+            this.state.terminal.send(value);
+        },
+
+        handleKeyDown: function(event) {
+            let keyCodesFuncs = {
+                Space: ' ',
+                ArrowUp: '\x1b[A',
+                ArrowDown: '\x1b[B',
+                ArrowRight: '\x1b[C',
+                ArrowLeft: '\x1b[D',
+            };
+
+            if (keyCodesFuncs[event.code]) {
+                this.sendToTerm(keyCodesFuncs[event.code]);
+            }
```

Comment:
Basically the idea here is, if users know the hotkeys and would like to use them to control playback, they can just focus the terminal and use them, or if they don't they can press on-screen buttons, which would generate them.